File: examples/node/scan-sbom.ts
import { readFile } from "node:fs/promises";

const apiBaseUrl = process.env.API_BASE_URL;
const serviceApiKey = process.env.SERVICE_API_KEY;
const sbomFile = process.env.SBOM_FILE ?? "./sbom.json";

if (!apiBaseUrl || !serviceApiKey) {
  throw new Error("Set API_BASE_URL and SERVICE_API_KEY");
}

const response = await fetch(`${apiBaseUrl}/scan-sbom`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${serviceApiKey}`,
    "Content-Type": "application/json",
    "X-Product-Name": "Demo Product",
    "X-Product-Version": "1.0.0",
    "Idempotency-Key": `node-scan-${Date.now()}`,
  },
  body: await readFile(sbomFile, "utf8"),
});

console.log(JSON.stringify(await response.json(), null, 2));

File: examples/python/scan_sbom.py
import json
import os
import urllib.request

api_base_url = os.environ["API_BASE_URL"]
service_api_key = os.environ["SERVICE_API_KEY"]
sbom_file = os.environ.get("SBOM_FILE", "./sbom.json")

with open(sbom_file, "rb") as file:
    body = file.read()

request = urllib.request.Request(
    f"{api_base_url}/scan-sbom",
    data=body,
    method="POST",
    headers={
        "Authorization": f"Bearer {service_api_key}",
        "Content-Type": "application/json",
        "X-Product-Name": "Demo Product",
        "X-Product-Version": "1.0.0",
        "Idempotency-Key": "python-scan-demo",
    },
)

with urllib.request.urlopen(request) as response:
    print(json.dumps(json.loads(response.read()), indent=2))

